src/component/noteCard/NoteCard.jsx

```diff
@@ -52,21 +101,43 @@ export const NoteCard = ({ item, archive }) => {
       </div>
 
       <div className="card-foot note-editing-option">
-        <button
-          className="btn btn-only-icon note-delete"
-          onClick={handleDelete}
-        >
-          <span className="fa-solid fa-trash"></span>
-        </button>
-        <button
-          className="btn btn-only-icon note-archive"
-          onClick={handleArchive}
-        >
-          <span className="fa-solid fa-box"></span>
-        </button>
-        <button className="btn btn-only-icon note-edit" onClick={handleEdit}>
-          <span className="fa-solid fa-pencil"></span>
-        </button>
+        {archive ? (
+          <>
+            <button
+              className="btn btn-only-icon note-delete"
+              onClick={handleDeleteArchive}
+            >
+              <span className="fa-solid fa-trash"></span>
+            </button>
+            <button
+              className="btn btn-only-icon note-archive"
+              onClick={handleArchiveRestore}
+            >
+              <span className="fa-solid fa-box-open"></span>
+            </button>
+          </>
+        ) : (
+          <>
+            <button
+              className="btn btn-only-icon note-delete"
+              onClick={handleDelete}
+            >
+              <span className="fa-solid fa-trash"></span>
+            </button>
+            <button
```

Comment:
Suggestion: these buttons are getting repeated, you can make a component and pass it onclick, and classes as a props